tests/ui/unnecessary_wrap.stderr

```diff
@@ -0,0 +1,53 @@
+error: this function returns unnecessarily wrapping data
+  --> $DIR/unnecessary_wrap.rs:8:1
+   |
+LL | / fn func1(a: bool, b: bool) -> Option<i32> {
+LL | |     if a && b {
+LL | |         return Some(42);
+LL | |     }
+...  |
+LL | |     }
+LL | | }
+   | |_^
+   |
+   = note: `-D clippy::unnecessary-wrap` implied by `-D warnings`
+help: factor this out to
+   |
+LL | fn func1(a: bool, b: bool) -> i32 {
+LL |     if a && b {
+LL |         return 42;
+LL |     }
+LL |     if a {
+LL |         Some(-1);
+ ...
```

Comment:
Since this is cut off, there is no test that **shows** that every `Some(..)` is replaced. Can you add such a test please.

Reply:
Changed accordingly by c17e278.